include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,217 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         * @param[in] x The coordinate where B-splines are evaluated.
+         * @return The index of the first B-spline which is evaluated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_basis(DSpan1D values, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates non-zero B-splines derivatives at a given coordinate
+         *
+         * The derivatives are computed for every B-splines with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are derivated.
+         * A spline approximation of a derivative at coordinate x is a linear
+         * combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] derivs The derivatives of the B-splines evaluated at coordinate x. It has to be a 1D mdspan with (degree+1) elements.
+         * @param[in] x The coordinate where B-splines derivatives are evaluated.
+         * @return The index of the first B-spline which is derivated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_deriv(DSpan1D derivs, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates non-zero B-splines values and \f$n\f$ derivatives at a given coordinate
+         *
+         * The values and derivatives are computed for every B-splines with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are derivated.
+         * A spline approximation of a derivative at coordinate x is a linear
+         * combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] derivs The values and \f$n\f$ derivatives of the B-splines evaluated at coordinate x. It has to be a 2D mdspan with (degree+1)*(n+1) elements.
+         * @param[in] x The coordinate where B-splines derivatives are evaluated.
+         * @param[in] n The number of derivatives to evaluate (in addition to the B-splines values themselves).
+         * @return The index of the first B-spline which is evaluated/derivated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type eval_basis_and_n_derivs(
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * The integral of each of the B-splines over their support within the domain on which this basis was defined.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a (nbasis) 1D mdspan.
```

Comment:
I am not sure about this syntax `(nbasis) 1D` for me it is not clear that `nbasis` refers to the size